src/libostree/ostree-sysroot-deploy.c

```diff
@@ -4099,10 +4099,30 @@ _ostree_sysroot_boot_complete (OstreeSysroot *self, GCancellable *cancellable, G
   if (!ot_openat_ignore_enoent (self->boot_fd, _OSTREE_FINALIZE_STAGED_FAILURE_PATH, &failure_fd,
                                 error))
     return FALSE;
-  // If we didn't find a failure log, then there's nothing to do right now.
-  // (Actually this unit shouldn't even be invoked, but we may do more in the future)
+  // If we didn't find a failure log, check for soft-reboot completion tasks
   if (failure_fd == -1)
-    return TRUE;
+    {
+      // Check if we just completed a soft-reboot and need to update /run/ostree-booted
+      if (g_file_test ("/run/ostree/nextroot-booted", G_FILE_TEST_EXISTS))
+        {
+          // We're completing a soft-reboot, simply move the nextroot-booted file to ostree-booted
+          if (rename ("/run/ostree/nextroot-booted", OTCORE_RUN_BOOTED) == 0)
+            {
+              g_print ("Updated /run/ostree-booted for soft-reboot completion\n");
```

Comment:
Probably just a `g_debug()`? Dunno if we need to print something every time here